http_parser.c

```diff
@@ -162,14 +162,40 @@ do {                                                                 \
 
 
 #define PROXY_CONNECTION "proxy-connection"
-#define CONNECTION "connection"
+#define ACCEPT "accept"
+#define CONTENT_ "content-"
 #define CONTENT_LENGTH "content-length"
-#define TRANSFER_ENCODING "transfer-encoding"
 #define UPGRADE "upgrade"
 #define CHUNKED "chunked"
 #define KEEP_ALIVE "keep-alive"
 #define CLOSE "close"
-
+#define IF_ "if-"
+#define PROXY_ "proxy-"
+#define X_FORWARDED_ "x-forwarded-"
+
+
+#define case_header(tag, string)                          \
+      case h_matching_##tag:                                    \
+        parser->index++;                                        \
+        if (parser->index > sizeof(string) - 1                  \
+            || c != string[parser->index]) {                    \
+          parser->header_state = h_general;                     \
+        } else if (parser->index == sizeof(string) - 2) {       \
```

Comment:
`sizeof(string) - 2` means that we have just read the last character (and that matched the expected character from `string[parser->index] == c` inside if above). So we are good to save the `header_state`. This is how headers are [matched today](https://github.com/nodejs/http-parser/blob/master/http_parser.c#L1368) and I used same logic to match other headers.